instrumentation/src/javatest/org/oppia/android/instrumentation/ExplorationPlayerTest.kt

```diff
@@ -0,0 +1,108 @@
+package org.oppia.android.instrumentation
+
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiCollection
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiScrollable
+import androidx.test.uiautomator.UiSelector
+import androidx.test.uiautomator.Until.hasObject
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+
+/** Tests for Explorations. */
+class ExplorationPlayerTest {
+  private val OPPIA_PACKAGE = "org.oppia.android"
+  private val LAUNCH_TIMEOUT = 30000L
+  private val TRANSITION_TIMEOUT = 5000L
+  private lateinit var device: UiDevice
+
+  @Before
+  fun startOppiaFromHomeScreen() {
+    // Initialize UiDevice instance
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+    // Start from the home screen
+    device.pressHome()
+
+    // Wait for launcher
+    val launcherPackage = getLauncherPackageName()
+    assertNotNull(launcherPackage)
+    device.wait(hasObject(By.pkg(launcherPackage)), LAUNCH_TIMEOUT)
+
+    // Launch the blueprint app
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    val intent = context.packageManager
+      .getLaunchIntentForPackage(OPPIA_PACKAGE)
+    intent!!.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK) // Clear out any previous instances
+    context.startActivity(intent)
+
+    // Wait for the app to appear
+    device.wait(hasObject(By.pkg(OPPIA_PACKAGE).depth(0)), LAUNCH_TIMEOUT)
+  }
+
+  @Test
+  fun testExploration_prototypeExploration_toolbarTitle_isDisplayedSuccessfully() {
+    NavigateToPrototypeExploration()
+    device.wait(
+      hasObject(
+        By.res("$OPPIA_PACKAGE:id/exploration_toolbar_title")
+      ),
+      TRANSITION_TIMEOUT
+    )
+    assertNotNull(device.findObject(By.res("$OPPIA_PACKAGE:id/exploration_toolbar_title")))
```

Comment:
Prefer using Truth assertions.

Reply:
Can't think of a way to use assertTrue here as there isn't any `isvisible` function for UiObject used assertEquals instead

Reply:
We can actually create our own extension for Truth (which is one of the major benefits of it over JUnit), but rather than doing that you can just do:

``assertThat(object).isNotNull()`` in place of the current line.

There are other benefits to Truth: it is easier to read since it uses a fluent pattern, and its error messages are better reported than JUnit's. We should still use it even if there aren't specific functions for these cases (Truth covers the same basic null/not-null/true/false cases JUnit's assertions do).

Reply:
Ah, That was quite confusing I thought you mentioned using AssertTure here.